shared/src/main/scala/mlscript/TyperDatatypes.scala

```diff
@@ -253,12 +254,38 @@ abstract class TyperDatatypes extends TyperHelpers { self: Typer =>
   case class TypeBounds(lb: SimpleType, ub: SimpleType)(val prov: TypeProvenance) extends SimpleType {
     def level: Int = lb.level max ub.level
     override def toString = s"$lb..$ub"
+    def && (that: TypeBounds, prov: TypeProvenance = noProv): TypeBounds =
```

Comment:
Question: is this actually still used?

Reply:
Seems not used from code lens, both`&&` and `||`

Reply:
Okay, I think we can remove it then.